sveltekit-authentication/index.md

```diff
@@ -0,0 +1,449 @@
+---
+title: 'Authentication in SvelteKit with Directus'
+description: 'How to authenticate users and integrate state managment with Cookies using SvelteKit SSR'
+author:
+  name: 'Eike Thies'
+  avatar_file_name: 'avatar.jpg'
+---
+
+
+## Overview
+
+In this Guide we will show how to set up a complete authentication and authorization mechanism in SvelteKit using Directus as the Provider. The user session will be persisted via Server-Side Cookies and evidently be used within the Directus SDK to make authenticated Calls. As an example we will create a new Role showing how authorization works.
+
+
+## Primer Authentication
+
+There are different ways how to store the user session. Here we will use the most secure form via Server-Side Cookies using the `Same-Site=Strict` attribute which will prevent CSRF Attacks. This means in order to be able to access the Directus Backend from the client directly, the Directus Backend will need to live on the same domain as the SvelteKit Frontend. For local development this will not matter. For more Information on this topic see https://andrewlock.net/understanding-samesite-cookies/
+
+
+## Adapt Directus Wrapper
+
+The following code examples will build up on the first Directus SvelteKit Blog Article. If you haven't seen it, you can check it out. Otherwise, with basic knowledge of SvelteKit you will be fine following along. First - let's change a few lines in our Directus SDK Wrapper, to utilize the Token and define a global Cookie Options Schema:
+
+
+```js [/libs/directus.js]
+import { createDirectus, rest  } from "@directus/sdk"; // [!code --]
+import { createDirectus, rest, authentication  } from "@directus/sdk"; // [!code ++]
+import { PUBLIC_APIURL } from '$env/static/public'; // [!code --]
+import { PUBLIC_APIURL,PUBLIC_COOKIE_DOMAIN } from '$env/static/public'; // [!code ++]
+
+function getDirectusInstance(fetch,token) {
+
+  const options = fetch ? { globals: { fetch } } : {};
```

Comment:
This doesn't need resolving by you but a note to self to run this through prettier before loading it in to be published. Spaces after commas is what caught my eye here.